[question] export-pkg failes on dependencies for a standalone application [conan 2]

### What is your question?

Hi,
I'm now trying to package an application that contains a single exe file. On the server it is build by cmake/msvc and then packaged and uploaded. Since it uses a static CRT due to a pdfium library the settings for build supplied by cmake are different than the one that are in the default profile when packaging. But since it is a standalone application and those settings are deleted in `package_id()` it should not matter, at least I thought.

The app builds fine with this, using the conan-cmake:
```pwsh
cmake -S . -B build --preset lx_current
cmake --build ./build --config Release
```
This is what conan-cmake outputs as a profile. The conan-cmake is slightly changed (should not matter, builds fine):
```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=17
compiler.runtime=static
compiler.runtime_type=Release
compiler.version=193
os=Windows
&:build_type=Debug
[conf]
tools.build:compiler_executables={'c': 'C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe', 'cpp': 'C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe'}
tools.cmake.cmake_layout:build_folder=D:/lexocad/Pdf2Image/code/build
tools.cmake.cmaketoolchain:generator=Visual Studio 17 2022

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows
```

Then I try:
```
conan export-pkg . --user lexocad-external --channel prebuild --version 1.0.9

...
pdfium/2020.9@pdf2image/prebuild: WARN: Can't find a 'pdfium/2020.9@pdf2image/prebuild' package binary '5cc52c47af3ba1977b0055afe30c9b905e30aa93' for the configuration:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows
```
Surprisingly, it is building a dependency graph even though the recipe states it is an application. It is built with one static and one header library. However, it fails to find the static one since the default profile differs(?). But I would expect that it wouldn't matter since the differences are in parts deleted in `package_id()`. I only need the compiler and build_type for the build step.

Ok then I tried to add `--skip-binaries` which then exploded on the error in boost. The recipe is the same as in the conancenter (downloaded from git today) but I reexported it under a different user/channel for further usage.
```
...

conanfile.py (pdf2image/1.0.9@lexocad-external/prebuild): Generator 'CMakeDeps' calling 'generate()'
conanfile.py (pdf2image/1.0.9@lexocad-external/prebuild): ERROR: Traceback (most recent call last):
  File "D:\lexocad\conan2_test\conan2env\Lib\site-packages\conan\tools\cmake\cmakedeps\templates\__init__.py", line 35, in render
    context = self.context
              ^^^^^^^^^^^^
  File "D:\lexocad\conan2_test\conan2env\Lib\site-packages\conan\tools\cmake\cmakedeps\templates\target_data.py", line 37, in context
    global_cpp = self._get_global_cpp_cmake()
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\lexocad\conan2_test\conan2env\Lib\site-packages\conan\tools\cmake\cmakedeps\templates\target_data.py", line 193, in _get_global_cpp_cmake
    return _TargetDataContext(global_cppinfo, pfolder_var_name, self._root_folder,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\lexocad\conan2_test\conan2env\Lib\site-packages\conan\tools\cmake\cmakedeps\templates\target_data.py", line 308, in __init__
    self.include_paths = join_paths(cpp_info.includedirs)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\lexocad\conan2_test\conan2env\Lib\site-packages\conan\tools\cmake\cmakedeps\templates\target_data.py", line 282, in join_paths
    assert os.path.isabs(p), "{} is not absolute".format(p)
           ^^^^^^^^^^^^^^^^
AssertionError: include is not absolute

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\lexocad\conan2_test\conan2env\Lib\site-packages\conans\client\generators\__init__.py", line 104, in write_generators
    generator.generate()
  File "D:\lexocad\conan2_test\conan2env\Lib\site-packages\conan\tools\cmake\cmakedeps\cmakedeps.py", line 47, in generate
    generator_files = self.content
                      ^^^^^^^^^^^^
  File "D:\lexocad\conan2_test\conan2env\Lib\site-packages\conan\tools\cmake\cmakedeps\cmakedeps.py", line 91, in content
    self._generate_files(require, dep, ret, find_module_mode=False)
  File "D:\lexocad\conan2_test\conan2env\Lib\site-packages\conan\tools\cmake\cmakedeps\cmakedeps.py", line 121, in _generate_files
    data_content = data_target.render()
                   ^^^^^^^^^^^^^^^^^^^^
  File "D:\lexocad\conan2_test\conan2env\Lib\site-packages\conan\tools\cmake\cmakedeps\templates\__init__.py", line 37, in render
    raise ConanException("error generating context for '{}': {}".format(self.conanfile, e))
conans.errors.ConanException: error generating context for 'boost/1.78.0@lexocad/prebuild': include is not absolute

ERROR: Error in generator 'CMakeDeps': error generating context for 'boost/1.78.0@lexocad/prebuild': include is not absolute
```

My wish for the resulting package would be a single exe file with os, and arch settings in the conaninfo for it is then packaged as a part of a different application that builds with different settings (more like the default profile). It was working for conan 1 though the requirements were set as private there. Why doesn't this work out of the box? I would like to have `settings = "os", "arch"` but then `conan install .` wouldn't work for the dependencies. Should I always set the `visible=False` in the `requires()` for applications? What is then the `package_type` for? Why doesn't it simply skip those dependencies?

The recipe for pdfium states `package_type = "static-library"`.
conan recipe for the app:
[conanfile.py](https://github.com/user-attachments/files/16085477/conanfile.py.txt)

conan version 2.4.1

Sorry for the long read.

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

--- a/test/integration/test_package_vendor.py
+++ b/test/integration/test_package_vendor.py
@@ -138,3 +138,42 @@ def test_vendor_dont_propagate_options():
     c.run("install consumer_shared --build=app/* -c tools.graph:vendor=build")
     c.assert_listed_binary({"pkga/0.1": ("55c609fe8808aa5308134cb5989d23d3caffccf2", "Cache"),
                             "app/0.1": ("da39a3ee5e6b4b0d3255bfef95601890afd80709", "Build")})
+
+
+def test_package_vendor_export_pkg():
+    c = TestClient()
+    app = textwrap.dedent("""
+        import os
+        from conan import ConanFile
+        from conan.tools.files import copy, save
+
+        class App(ConanFile):
+            name = "app"
+            version = "0.1"
+            package_type = "application"
+            vendor = True
+            requires = "pkga/0.1"
+            def package(self):
+                copy(self, "*", src=self.dependencies["pkga"].package_folder,
+                     dst=self.package_folder)
+                save(self, os.path.join(self.package_folder, "app.exe"), "app")
+            """)
+
+    c.save({"pkga/conanfile.py": GenConanfile("pkga", "0.1").with_package_type("shared-library")
+                                                            .with_package_file("pkga.dll", "dll"),
+            "app/conanfile.py": app
+            })
+    c.run("create pkga")
+    c.run("build app")  # -c tools.graph:vendor=build will be automatic
+    c.run("export-pkg app")
+    assert "pkga/0.1" in c.out  # In the export-pkg process, dependencies are still needed
+    assert "conanfile.py (app/0.1): package(): Packaged 1 '.dll' file: pkga.dll" in c.out
+
+    # we can safely remove pkga, once the package is created, it can work without deps
+    c.run("remove pkg* -c")
+    c.run("list app:*")
+    assert "pkga" not in c.out  # The binary doesn't depend on pkga
+    c.run("install --requires=app/0.1 --deployer=full_deploy")
+    assert "pkga" not in c.out
+    assert c.load("full_deploy/host/app/0.1/app.exe") == "app"
+    assert c.load("full_deploy/host/app/0.1/pkga.dll") == "dll"

